ec_grasp_planner/src/handarm_parameters.py

```diff
@@ -66,71 +65,64 @@ class RBOHandP24WAM(RBOHand2):
     def __init__(self, **kwargs):
         super(RBOHandP24WAM, self).__init__()
 
-        # does the robot support impedance control
-        self['isForceControllerAvailable'] = True
-
         # you can define a default strategy for all objects by setting the second field to  'object'
         # for object-specific strategies set it to the object label
 
         # transformation between object frame and hand palm frame above the object- should not be changed per object
         # please don't set x and y position, this should be done in pre_grasp_transform
         # the convention at our lab is: x along the fingers and z normal on the palm.
         # please follow the same convention
-        self['surface_grasp']['object']['hand_transform'] = tra.concatenate_matrices(tra.translation_matrix([0.0, 0.0, 0.3]),
+        self['SurfaceGrasp']['object']['hand_transform'] = tra.concatenate_matrices(tra.translation_matrix([0.0, 0.0, 0.3]),
                                                                                 tra.concatenate_matrices(
                                                                                     tra.rotation_matrix(
                                                                                         math.radians(90.), [0, 0, 1]),
                                                                                     tra.rotation_matrix(
                                                                                         math.radians(180.), [1, 0, 0])))
 
-        # position of hand relative to the object before and at grasping
-        self['surface_grasp']['object']['pregrasp_transform'] = tra.concatenate_matrices(
-            tra.translation_matrix([0.0, 0.0, 0.0]), tra.rotation_matrix(math.radians(10.0), [0, 1, 0]))
+        # transformation between the control frame of the hand and the frame in which the hand transform is defined
+        # this is needed for the PISA hand to enforce the grasping signature
+        self['SurfaceGrasp']['object']['ee_in_goal_frame'] = tra.translation_matrix([0.0, 0.0, 0.0])
```

Comment:
If this is only needed for the PISA hand, why is it defined in RBOHandP24WAM?

Reply:
In planner.py we are doing:
 ```
elif grasp_type == 'SurfaceGrasp':
        return ((object_pose.dot(params['hand_transform'])).dot(params['pre_approach_transform'])).dot(params['ee_in_goal_frame'])
```

So, we that's why `ee_in_goal_frame` had to exist for all hand-robot combinations. But we could do a check like ```if 'pisa' in handarm_type``` instead if you prefer...

Reply:
okay, that makes sense. I think we should keep it then.